test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2178,6 +2179,14 @@ public static <T> T safeGet(Future<T> future) {
         }
     }
 
+    public static Exception safeAwaitFailure(SubscribableListener<?> listener) {
```

Comment:
Nit: I think we can have a variant of this method that takes `PlainActionFuture` in a future PR.